### DISCUSSION TITLE
Can I contribute an ADC0834 implementation?

### DISCUSSION THREAD

**Discussion by hankehly** [gpiozero/gpiozero#1094] [Ideas]
### Device name
ADC0834 analog to digital converter

### Protocol
Unsure (uses GPIO pins)

### URL of example product
https://docs.sunfounder.com/projects/raphael-kit/en/latest/components/component_adc0834.html

### Further information
* [Data sheet](https://www.ti.com/lit/ds/symlink/adc0831-n.pdf)
* [A blog article I wrote](https://medium.com/@hankehly/adc0834-a-step-by-step-guide-to-analog-to-digital-conversion-fa502eacc0c9) about the steps to read a value in Python

### Please provide example code for using this device with Python, using another library

I wrote my own module and, if it's OK with the maintainers, would be willing to spruce it up and contribute to this project for others to use.
https://github.com/hankehly/raspberry-pi-sandbox/blob/main/raspberry_pi_sandbox/ADC0834.py

```python
adc = ADC0834(cs=11, clk=13, dio=15)
adc.setup()
adc.read(channel=0)
```

**Comment by bennuttall** [comment]
It may be possible to integrate this into gpiozero itself but as you can see from the existing PRs, we don't really have time to do the work to bring this kind of work in.

I would suggest you write a version of that code using gpiozero instead of RPi.GPIO, but in the same way, i.e. you import stuff from gpiozero and write your own class. I'd copy the API from the MCP3xxx series which you'll find here: https://github.com/gpiozero/gpiozero/blob/master/gpiozero/spi_devices.py

You could then share your version for others to use or even publish it as a separate module on PyPI.

**Comment by lurch** [comment]
https://www.ti.com/product/ADC0834-N says that it uses SPI, so you _might_ want at look at using https://gpiozero.readthedocs.io/en/latest/api_spi.html
